smac/facade/smac_facade.py

```diff
@@ -287,170 +357,145 @@ def __init__(self,
+            'stats':self.stats,
+            'traj_logger':traj_logger,
+            'rng':rng,
+            'instances':scenario.train_insts,
+            'cutoff':scenario.cutoff,
+            'deterministic':scenario.deterministic,
+            'run_obj_time':scenario.run_obj == "runtime",
+            'always_race_against':scenario.cs.get_default_configuration()
+                                   if scenario.always_race_default else None,
+            'use_ta_time_bound':scenario.use_ta_time,
+            'instance_specifics':scenario.instance_specific,
+            'minR':scenario.minR,
+            'maxR':scenario.maxR,
+            'adaptive_capping_slackfactor':scenario.intens_adaptive_capping_slackfactor,
+            'min_chall':scenario.intens_min_chall
+            }
+        intensifier_def_kwargs.update(intensifier_kwargs)
         if intensifier is None:
-            intensifier = Intensifier(tae_runner=tae_runner,
-                                      stats=self.stats,
-                                      traj_logger=traj_logger,
-                                      rng=rng,
-                                      instances=scenario.train_insts,
-                                      cutoff=scenario.cutoff,
-                                      deterministic=scenario.deterministic,
-                                      run_obj_time=scenario.run_obj == "runtime",
-                                      always_race_against=scenario.cs.get_default_configuration()
-                                      if scenario.always_race_default else None,
-                                      use_ta_time_bound=scenario.use_ta_time,
-                                      instance_specifics=scenario.instance_specific,
-                                      minR=scenario.minR,
-                                      maxR=scenario.maxR,
-                                      adaptive_capping_slackfactor=scenario.intens_adaptive_capping_slackfactor,
-                                      min_chall=scenario.intens_min_chall)
-        # inject deps if necessary
-        if intensifier.tae_runner is None:
-            intensifier.tae_runner = tae_runner
-        if intensifier.stats is None:
-            intensifier.stats = self.stats
-        if intensifier.traj_logger is None:
-            intensifier.traj_logger = traj_logger
+            intensifier = Intensifier(**intensifier_def_kwargs)
+        elif inspect.isclass(intensifier):
+            intensifier = intensifier(**intensifier_def_kwargs)
+        elif isinstance(intensifier, Intensifier):
+            if intensifier.tae_runner is None:
+                intensifier.tae_runner = tae_runner
+            if intensifier.stats is None:
+                intensifier.stats = self.stats
+            if intensifier.traj_logger is None:
+                intensifier.traj_logger = traj_logger
+        #TODO: no further check since we have no abstract intensifier
```

Comment:
How about an else which raises an exception?

Reply:
No, if someone has implemented his/her own intensification approach (with the same API as we use), he/she should be able to still use the SMAC facade

Reply:
But shouldn't that be covered by the `elif isinstance(...)`? Otherwise, there's no/little guarantee that the objects actually fit together.